site/app/controllers/admin/ReportController.php

```diff
@@ -589,6 +589,35 @@ private function getLateStatusMessage(LateDayInfo $ldi) {
         }
     }
 
+
+    #[Route("/courses/{_semester}/{_course}/reports/rainbow_grades_customization_build", methods: ["POST"])]
+    public function generateCustomizationBuild(): void {
+        // Check if this is an AJAX request
+        if (!empty($_SERVER['HTTP_X_REQUESTED_WITH']) && strtolower($_SERVER['HTTP_X_REQUESTED_WITH']) == 'xmlhttprequest') {
+            // Build a new model, pull in defaults for the course
+            $customization = new RainbowCustomization($this->core);
+            $customization->buildCustomization();
+            if (isset($_POST["json_string"])) {
+                try {
+                    $customization->processForm();
+                    die("success");
+                }
+                catch (\Exception $e) {
+                    die("error");
+                }
+            }
+            else {
+                die("error: no json_string provided");
+            }
+        }
+        else {
+            die("error: not an AJAX request");
```

Comment:
It would be better to use json success and fail responses instead of just exiting immediately.